Disable stoping of addon after finding vulnerability

**Is your feature request related to a problem? Please describe.**
The current architecture of addon is such that it will stop sending more malicious requests after one vulnerability is reported. We need to enhance the addon to continue sending more requests and this will be based on the flag from the UI. If the flag is true then continue sending the requests.

**Describe the solution you'd like**
File upload addon has a ui:
![image](https://github.com/SasanLabs/owasp-zap-fileupload-addon/assets/13570884/956a0734-0b0a-4d86-a248-d60061e67d20)

We will add a new checkbox which is to continue running after finding the vulnerability. This way it will help users to find all the vulnerabilities in there web application.

@preetkaran20, when running **_./gradlew install_**, I am getting the following:

```
2023-10-06T09:24:03.155+0300 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > An error occurred while installing the add-on: java.net.ConnectException: Connection refused: connect
2023-10-06T09:24:03.155+0300 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
2023-10-06T09:24:03.155+0300 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Try:
2023-10-06T09:24:03.155+0300 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Run with --stacktrace option to get the stack trace.  Run with --scan to get full insights.
```

The output above is with running the above command with --debug.
When I run it with --stacktrace, I get:

```

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':installZapAddOn'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:187)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:268)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:185)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:173)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
        at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:408)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:395)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:388)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:374)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: org.zaproxy.gradle.addon.AddOnPluginException: An error occurred while installing the add-on: java.net.ConnectException: Connection refused: connect
        at org.zaproxy.gradle.addon.misc.InstallAddOn.start(InstallAddOn.java:59)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$2.run(ExecuteActionsTaskExecuter.java:498)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:56)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$run$1(DefaultBuildOperationExecutor.java:71)
        at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.runWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:45)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:71)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:483)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:466)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$300(ExecuteActionsTaskExecuter.java:105)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:270)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:248)
        at org.gradle.internal.execution.steps.ExecuteStep.executeInternal(ExecuteStep.java:83)
        at org.gradle.internal.execution.steps.ExecuteStep.access$000(ExecuteStep.java:37)
        at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:50)
        at org.gradle.internal.execution.steps.ExecuteStep$1.call(ExecuteStep.java:47)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
        at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:47)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:37)
        at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:68)
        at org.gradle.internal.execution.steps.RemovePreviousOutputsStep.execute(RemovePreviousOutputsStep.java:38)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:50)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:36)
        at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:41)
        at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:74)
        at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:55)
        at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:51)
        at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:29)
        at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:54)
        at org.gradle.internal.execution.steps.CaptureStateAfterExecutionStep.execute(CaptureStateAfterExecutionStep.java:35)
        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:60)
        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:27)
        at org.gradle.internal.execution.steps.BuildCacheStep.executeWithoutCache(BuildCacheStep.java:174)
        at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:74)
        at org.gradle.internal.execution.steps.BuildCacheStep.execute(BuildCacheStep.java:45)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:40)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:29)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:36)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:22)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:99)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:92)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:52)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:36)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:84)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:41)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:37)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:27)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:91)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:49)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:78)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:49)
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:105)
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:50)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:86)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:86)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:32)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:38)
        at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:43)
        at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:31)
        at org.gradle.internal.execution.steps.AssignWorkspaceStep.lambda$execute$0(AssignWorkspaceStep.java:40)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution$2.withWorkspace(ExecuteActionsTaskExecuter.java:283)
        at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:40)
        at org.gradle.internal.execution.steps.AssignWorkspaceStep.execute(AssignWorkspaceStep.java:30)
        at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:37)
        at org.gradle.internal.execution.steps.IdentityCacheStep.execute(IdentityCacheStep.java:27)
        at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:49)
        at org.gradle.internal.execution.steps.IdentifyStep.execute(IdentifyStep.java:35)
        at org.gradle.internal.execution.impl.DefaultExecutionEngine$1.execute(DefaultExecutionEngine.java:76)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:184)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:173)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:109)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:200)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:195)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:75)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$3.execute(DefaultBuildOperationRunner.java:68)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:153)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:68)
        at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:62)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.lambda$call$2(DefaultBuildOperationExecutor.java:76)
        at org.gradle.internal.operations.UnmanagedBuildOperationWrapper.callWithUnmanagedSupport(UnmanagedBuildOperationWrapper.java:54)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:76)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:408)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:395)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:388)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:374)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: org.zaproxy.clientapi.core.ClientApiException: java.net.ConnectException: Connection refused: connect
        at org.zaproxy.clientapi.core.ClientApi.callApiDom(ClientApi.java:366)
        at org.zaproxy.clientapi.core.ClientApi.callApi(ClientApi.java:350)
        at org.zaproxy.clientapi.gen.Autoupdate.installLocalAddon(Autoupdate.java:152)
        at org.zaproxy.gradle.addon.misc.InstallAddOn.start(InstallAddOn.java:53)
        ... 122 more
Caused by: java.net.ConnectException: Connection refused: connect
        at org.zaproxy.clientapi.core.ClientApi.getConnectionInputStream(ClientApi.java:398)
        at org.zaproxy.clientapi.core.ClientApi.callApiDom(ClientApi.java:364)
        ... 125 more
```

How can I overcome this? It seems to be related to the connection not succeeding.

--- a/src/main/java/org/sasanlabs/fileupload/attacks/FileUploadAttackExecutor.java
+++ b/src/main/java/org/sasanlabs/fileupload/attacks/FileUploadAttackExecutor.java
@@ -72,16 +72,14 @@ public class FileUploadAttackExecutor {
 
     public boolean executeAttack() throws FileUploadException {
 
-        Boolean shouldSendRequestsAfterFindingVulnerability = FileUploadConfiguration
-                .getInstance()
-                .getSendRequestsAfterFindingVulnerability();
+        Boolean shouldSendRequestsAfterFindingVulnerability =
+                FileUploadConfiguration.getInstance().getSendRequestsAfterFindingVulnerability();
 
         for (AttackVector attackVector : attackVectors) {
             if (this.fileUploadScanRule.isStop()) {
                 return false;
             } else {
-                if (attackVector.execute(this) &&
-                        !shouldSendRequestsAfterFindingVulnerability) {
+                if (attackVector.execute(this) && !shouldSendRequestsAfterFindingVulnerability) {
                     return true;
                 }
             }

--- a/src/main/java/org/sasanlabs/fileupload/attacks/FileUploadAttackExecutor.java
+++ b/src/main/java/org/sasanlabs/fileupload/attacks/FileUploadAttackExecutor.java
@@ -29,6 +29,7 @@ import org.sasanlabs.fileupload.attacks.rce.php.ImageWithPHPSnippetFileUpload;
 import org.sasanlabs.fileupload.attacks.rce.php.SimplePHPFileUpload;
 import org.sasanlabs.fileupload.attacks.xss.HtmlFileUpload;
 import org.sasanlabs.fileupload.attacks.xss.SVGFileUpload;
+import org.sasanlabs.fileupload.configuration.FileUploadConfiguration;
 import org.sasanlabs.fileupload.exception.FileUploadException;
 
 /**
@@ -70,11 +71,17 @@ public class FileUploadAttackExecutor {
     }
 
     public boolean executeAttack() throws FileUploadException {
+
+        Boolean shouldSendRequestsAfterFindingVulnerability = FileUploadConfiguration
+                .getInstance()
+                .getSendRequestsAfterFindingVulnerability();
+
         for (AttackVector attackVector : attackVectors) {
             if (this.fileUploadScanRule.isStop()) {
                 return false;
             } else {
-                if (attackVector.execute(this)) {
+                if (attackVector.execute(this) &&
+                        !shouldSendRequestsAfterFindingVulnerability) {
                     return true;
                 }
             }

--- a/src/main/java/org/sasanlabs/fileupload/attacks/FileUploadAttackExecutor.java
+++ b/src/main/java/org/sasanlabs/fileupload/attacks/FileUploadAttackExecutor.java
@@ -1,5 +1,5 @@
 /**
- * Copyright 2021 SasanLabs
+ * Copyright 2023 SasanLabs
  *
  * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
  * except in compliance with the License. You may obtain a copy of the License at
@@ -29,6 +29,7 @@ import org.sasanlabs.fileupload.attacks.rce.php.ImageWithPHPSnippetFileUpload;
 import org.sasanlabs.fileupload.attacks.rce.php.SimplePHPFileUpload;
 import org.sasanlabs.fileupload.attacks.xss.HtmlFileUpload;
 import org.sasanlabs.fileupload.attacks.xss.SVGFileUpload;
+import org.sasanlabs.fileupload.configuration.FileUploadConfiguration;
 import org.sasanlabs.fileupload.exception.FileUploadException;
 
 /**
@@ -70,8 +71,13 @@ public class FileUploadAttackExecutor {
     }
 
     public boolean executeAttack() throws FileUploadException {
+        Boolean shouldSendRequestsAfterFindingVulnerability = FileUploadConfiguration
+                .getInstance()
+                .getSendRequestsAfterFindingVulnerability();
+
         for (AttackVector attackVector : attackVectors) {
-            if (this.fileUploadScanRule.isStop()) {
+            if (!shouldSendRequestsAfterFindingVulnerability &&
+                    this.fileUploadScanRule.isStop()) {
                 return false;
             } else {
                 if (attackVector.execute(this)) {

--- a/src/main/java/org/sasanlabs/fileupload/attacks/FileUploadAttackExecutor.java
+++ b/src/main/java/org/sasanlabs/fileupload/attacks/FileUploadAttackExecutor.java
@@ -29,7 +29,6 @@ import org.sasanlabs.fileupload.attacks.rce.php.ImageWithPHPSnippetFileUpload;
 import org.sasanlabs.fileupload.attacks.rce.php.SimplePHPFileUpload;
 import org.sasanlabs.fileupload.attacks.xss.HtmlFileUpload;
 import org.sasanlabs.fileupload.attacks.xss.SVGFileUpload;
-import org.sasanlabs.fileupload.configuration.FileUploadConfiguration;
 import org.sasanlabs.fileupload.exception.FileUploadException;
 
 /**
@@ -71,13 +70,8 @@ public class FileUploadAttackExecutor {
     }
 
     public boolean executeAttack() throws FileUploadException {
-        Boolean shouldSendRequestsAfterFindingVulnerability = FileUploadConfiguration
-                .getInstance()
-                .getSendRequestsAfterFindingVulnerability();
-
         for (AttackVector attackVector : attackVectors) {
-            if (!shouldSendRequestsAfterFindingVulnerability &&
-                    this.fileUploadScanRule.isStop()) {
+            if (this.fileUploadScanRule.isStop()) {
                 return false;
             } else {
                 if (attackVector.execute(this)) {

--- a/src/main/java/org/sasanlabs/fileupload/configuration/FileUploadConfiguration.java
+++ b/src/main/java/org/sasanlabs/fileupload/configuration/FileUploadConfiguration.java
@@ -109,6 +109,10 @@ public class FileUploadConfiguration extends VersionedAbstractParam {
                         parseResponseEndIdentifier);
     }
 
+    public Boolean getSendRequestsAfterFindingVulnerability() {
+        return sendRequestsAfterFindingVulnerability;
+    }
+
     public void setSendRequestsAfterFindingVulnerability(boolean shouldSendRequestsAfterFindingVulnerability) {
         sendRequestsAfterFindingVulnerability = shouldSendRequestsAfterFindingVulnerability;
         this.getConfig()

--- a/src/main/java/org/sasanlabs/fileupload/configuration/FileUploadConfiguration.java
+++ b/src/main/java/org/sasanlabs/fileupload/configuration/FileUploadConfiguration.java
@@ -1,5 +1,5 @@
 /**
- * Copyright 2021 SasanLabs
+ * Copyright 2023 SasanLabs
  *
  * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
  * except in compliance with the License. You may obtain a copy of the License at
@@ -39,11 +39,15 @@ public class FileUploadConfiguration extends VersionedAbstractParam {
             PARAM_BASE_KEY + ".parseresponse.startidentifier";
     private static final String PARAM_PARSE_RESPONSE_CONFIGURATION_END_IDENTIFIER =
             PARAM_BASE_KEY + ".parseresponse.endidentifier";
+    private static final String PARAM_SEND_REQUESTS_AFTER_FINDING_VULNERABILITY_IDENTIFIER =
+            PARAM_BASE_KEY + ".sendrequests";
 
     private String staticLocationURIRegex;
     private String dynamicLocationURIRegex;
     private String parseResponseStartIdentifier;
     private String parseResponseEndIdentifier;
+
+    private Boolean sendRequestsAfterFindingVulnerability;
 
     private static volatile FileUploadConfiguration fileUploadConfiguration;
 
@@ -105,6 +109,14 @@ public class FileUploadConfiguration extends VersionedAbstractParam {
                         parseResponseEndIdentifier);
     }
 
+    public void setSendRequestsAfterFindingVulnerability(boolean shouldSendRequestsAfterFindingVulnerability) {
+        sendRequestsAfterFindingVulnerability = shouldSendRequestsAfterFindingVulnerability;
+        this.getConfig()
+                .setProperty(
+                        PARAM_SEND_REQUESTS_AFTER_FINDING_VULNERABILITY_IDENTIFIER,
+                        shouldSendRequestsAfterFindingVulnerability);
+    }
+
     @Override
     protected String getConfigVersionKey() {
         return CONFIG_VERSION_KEY;
@@ -125,6 +137,8 @@ public class FileUploadConfiguration extends VersionedAbstractParam {
                 getConfig().getString(PARAM_PARSE_RESPONSE_CONFIGURATION_START_IDENTIFIER));
         this.setParseResponseEndIdentifier(
                 getConfig().getString(PARAM_PARSE_RESPONSE_CONFIGURATION_END_IDENTIFIER));
+        this.setSendRequestsAfterFindingVulnerability(
+                getConfig().getBoolean(PARAM_SEND_REQUESTS_AFTER_FINDING_VULNERABILITY_IDENTIFIER));
     }
 
     @Override

--- a/src/main/java/org/sasanlabs/fileupload/i18n/FileUploadI18n.java
+++ b/src/main/java/org/sasanlabs/fileupload/i18n/FileUploadI18n.java
@@ -1,5 +1,5 @@
 /**
- * Copyright 2021 SasanLabs
+ * Copyright 2023 SasanLabs
  *
  * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
  * except in compliance with the License. You may obtain a copy of the License at

--- a/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
+++ b/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
@@ -81,7 +81,9 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
 
     private JCheckBox buildSendRequestsAfterFindingVulnerabilityCheckbox() {
         sendRequestsAfterFindingVulnerability =
-                new JCheckBox(FileUploadI18n.getMessage("fileupload.settings.checkbox.sendrequestsaftervulnerability"));
+                new JCheckBox(
+                        FileUploadI18n.getMessage(
+                                "fileupload.settings.checkbox.sendrequestsaftervulnerability"));
         sendRequestsAfterFindingVulnerability.setSelected(false);
         return sendRequestsAfterFindingVulnerability;
     }

--- a/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
+++ b/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
@@ -1,5 +1,5 @@
 /**
- * Copyright 2021 SasanLabs
+ * Copyright 2023 SasanLabs
  *
  * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
  * except in compliance with the License. You may obtain a copy of the License at
@@ -21,6 +21,7 @@ import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
 import javax.swing.BoxLayout;
 import javax.swing.JButton;
+import javax.swing.JCheckBox;
 import javax.swing.JLabel;
 import javax.swing.JPanel;
 import javax.swing.JScrollPane;
@@ -52,6 +53,8 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
     private JTextField parseResponseStartIdentifier;
     private JTextField parseResponseEndIdentifier;
 
+    private JCheckBox sendRequestsAfterFindingVulnerability;
+
     public FileUploadOptionsPanel() {
         super();
         this.setName(FileUploadI18n.getMessage("fileupload.settings.title"));
@@ -72,7 +75,14 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
 
     private void init(JPanel settingsPanel) {
         settingsPanel.add(uriLocatorConfiguration());
+        settingsPanel.add(buildSendRequestsAfterFindingVulnerabilityCheckbox());
         footerPanel.add(getResetButton());
+    }
+
+    private JCheckBox buildSendRequestsAfterFindingVulnerabilityCheckbox() {
+        sendRequestsAfterFindingVulnerability =
+                new JCheckBox("Continue Sending Requests After Vulnerability Reported");
+        return sendRequestsAfterFindingVulnerability;
     }
 
     private JButton getResetButton() {

--- a/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
+++ b/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
@@ -88,8 +88,8 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
                                 "fileupload.settings.checkbox.sendrequestsaftervulnerability"));
 
         sendRequestsAfterFindingVulnerability = new JCheckBox();
-        sendRequestsAfterFindingVulnerabilityLabel.add(sendRequestsAfterFindingVulnerability);
         sendRequestsAfterFindingVulnerabilityPanel.add(sendRequestsAfterFindingVulnerabilityLabel);
+        sendRequestsAfterFindingVulnerabilityPanel.add(sendRequestsAfterFindingVulnerability);
 
         return sendRequestsAfterFindingVulnerabilityPanel;
     }

--- a/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
+++ b/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
@@ -82,6 +82,7 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
     private JCheckBox buildSendRequestsAfterFindingVulnerabilityCheckbox() {
         sendRequestsAfterFindingVulnerability =
                 new JCheckBox("Continue Sending Requests After Vulnerability Reported");
+        sendRequestsAfterFindingVulnerability.setSelected(false);
         return sendRequestsAfterFindingVulnerability;
     }
 
@@ -235,6 +236,7 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
         dynamicLocationConfigurationURIRegex.setText("");
         parseResponseStartIdentifier.setText("");
         parseResponseEndIdentifier.setText("");
+        sendRequestsAfterFindingVulnerability.setSelected(false);
     }
 
     @Override

--- a/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
+++ b/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
@@ -79,13 +79,17 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
         footerPanel.add(getResetButton());
     }
 
-    private JCheckBox buildSendRequestsAfterFindingVulnerabilityCheckbox() {
-        sendRequestsAfterFindingVulnerability =
-                new JCheckBox(
+    private JLabel buildSendRequestsAfterFindingVulnerabilityCheckbox() {
+        JLabel label =
+                new JLabel(
                         FileUploadI18n.getMessage(
                                 "fileupload.settings.checkbox.sendrequestsaftervulnerability"));
-        sendRequestsAfterFindingVulnerability.setSelected(false);
-        return sendRequestsAfterFindingVulnerability;
+
+        sendRequestsAfterFindingVulnerability = new JCheckBox();
+
+        label.add(sendRequestsAfterFindingVulnerability);
+
+        return label;
     }
 
     private JButton getResetButton() {

--- a/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
+++ b/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
@@ -81,7 +81,7 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
 
     private JCheckBox buildSendRequestsAfterFindingVulnerabilityCheckbox() {
         sendRequestsAfterFindingVulnerability =
-                new JCheckBox("Continue Sending Requests After Vulnerability Reported");
+                new JCheckBox(FileUploadI18n.getMessage("fileupload.settings.checkbox.sendrequestsaftervulnerability"));
         sendRequestsAfterFindingVulnerability.setSelected(false);
         return sendRequestsAfterFindingVulnerability;
     }

--- a/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
+++ b/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
@@ -251,6 +251,8 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
         parseResponseStartIdentifier.setText(
                 fileUploadConfiguration.getParseResponseStartIdentifier());
         parseResponseEndIdentifier.setText(fileUploadConfiguration.getParseResponseEndIdentifier());
+        sendRequestsAfterFindingVulnerability.setSelected(
+                fileUploadConfiguration.getSendRequestsAfterFindingVulnerability());
     }
 
     @Override

--- a/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
+++ b/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
@@ -287,7 +287,7 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
     }
 
     @Override
-    public void saveParam(Object optionParams) throws Exception {
+    public void saveParam(Object optionParams) {
         FileUploadConfiguration fileUploadConfiguration =
                 ((OptionsParam) optionParams).getParamSet(FileUploadConfiguration.class);
         fileUploadConfiguration.setStaticLocationURIRegex(

--- a/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
+++ b/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
@@ -80,16 +80,15 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
     }
 
     private JLabel buildSendRequestsAfterFindingVulnerabilityCheckbox() {
-        JLabel label =
+        JLabel SendRequestsAfterFindingVulnerabilityLabel =
                 new JLabel(
                         FileUploadI18n.getMessage(
                                 "fileupload.settings.checkbox.sendrequestsaftervulnerability"));
 
         sendRequestsAfterFindingVulnerability = new JCheckBox();
+        SendRequestsAfterFindingVulnerabilityLabel.add(sendRequestsAfterFindingVulnerability);
 
-        label.add(sendRequestsAfterFindingVulnerability);
-
-        return label;
+        return SendRequestsAfterFindingVulnerabilityLabel;
     }
 
     private JButton getResetButton() {

--- a/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
+++ b/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
@@ -79,16 +79,19 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
         footerPanel.add(getResetButton());
     }
 
-    private JLabel buildSendRequestsAfterFindingVulnerabilityCheckbox() {
-        JLabel SendRequestsAfterFindingVulnerabilityLabel =
+    private JPanel buildSendRequestsAfterFindingVulnerabilityCheckbox() {
+        JPanel sendRequestsAfterFindingVulnerabilityPanel = new JPanel();
+        sendRequestsAfterFindingVulnerabilityPanel.setLayout(new FlowLayout(FlowLayout.LEFT));
+        JLabel sendRequestsAfterFindingVulnerabilityLabel =
                 new JLabel(
                         FileUploadI18n.getMessage(
                                 "fileupload.settings.checkbox.sendrequestsaftervulnerability"));
 
         sendRequestsAfterFindingVulnerability = new JCheckBox();
-        SendRequestsAfterFindingVulnerabilityLabel.add(sendRequestsAfterFindingVulnerability);
+        sendRequestsAfterFindingVulnerabilityLabel.add(sendRequestsAfterFindingVulnerability);
+        sendRequestsAfterFindingVulnerabilityPanel.add(sendRequestsAfterFindingVulnerabilityLabel);
 
-        return SendRequestsAfterFindingVulnerabilityLabel;
+        return sendRequestsAfterFindingVulnerabilityPanel;
     }
 
     private JButton getResetButton() {

--- a/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
+++ b/src/main/java/org/sasanlabs/fileupload/ui/FileUploadOptionsPanel.java
@@ -298,5 +298,7 @@ public class FileUploadOptionsPanel extends AbstractParamPanel {
                 this.parseResponseStartIdentifier.getText());
         fileUploadConfiguration.setParseResponseEndIdentifier(
                 this.parseResponseEndIdentifier.getText());
+        fileUploadConfiguration.setSendRequestsAfterFindingVulnerability(
+                this.sendRequestsAfterFindingVulnerability.isSelected());
     }
 }

--- a/src/main/resources/org/sasanlabs/fileupload/i18n/Messages.properties
+++ b/src/main/resources/org/sasanlabs/fileupload/i18n/Messages.properties
@@ -127,4 +127,4 @@ fileupload.scanner.vulnerability.htaccessFile.soln=Follow the suggestions mentio
 3. https://www.youtube.com/watch?v=CmF9sEyKZNo \
 4. https://cwe.mitre.org/data/definitions/434.html
 
-fileupload.settings.checkbox.sendrequestsaftervulnerability=Continue Sending Requests After Vulnerability Reported
+fileupload.settings.checkbox.sendrequestsaftervulnerability=Keep exploiting after discovery

--- a/src/main/resources/org/sasanlabs/fileupload/i18n/Messages.properties
+++ b/src/main/resources/org/sasanlabs/fileupload/i18n/Messages.properties
@@ -126,3 +126,5 @@ fileupload.scanner.vulnerability.htaccessFile.soln=Follow the suggestions mentio
 2. https://owasp.org/www-community/vulnerabilities/Unrestricted_File_Upload \
 3. https://www.youtube.com/watch?v=CmF9sEyKZNo \
 4. https://cwe.mitre.org/data/definitions/434.html
+
+fileupload.settings.checkbox.sendrequestsaftervulnerability=Continue Sending Requests After Vulnerability Reported

